src/test/groovy/com/synopsys/integration/utility/VersionUtilityTest.groovy

```diff
@@ -170,6 +170,11 @@ class VersionUtilityTest {
         assertEquals('project-name-9.10.1-SIGQA1', versionUtility.removeBranchNameFromVersion(currentVersion))
         assertEquals('project-name-9.10.1-SIGQA2-dev_name.branch-name', versionUtility.calculateNextQAVersionDetect(currentVersion, 'origin/dev_name/branch-name'))
 
+        currentVersion = 'project-name-9.10.1-SIGQA1-dev_name.branch-name-SNAPSHOT'
+        assertEquals('project-name-9.10.1-SIGQA1-SNAPSHOT', versionUtility.removeBranchNameFromVersion(currentVersion))
+        assertEquals('project-name-9.10.1-SIGQA1', versionUtility.calculateNextQAVersionDetect(currentVersion, 'origin/master'))
```

Comment:
```
assertEquals('project-name-9.10.1-SIGQA1', versionUtility.calculateNextQAVersionDetect(currentVersion, 'origin/master'))
```

I would think that when calling calculateNextQAVersionDetect, if the version has SIGQA1 in it, then the return would be SIGQA2, even when the branch is 'origin/master'. It looks like anytime the input version contains SIGQA and SNAPSHOT, and the branch is 'origin/master', the SIGQA value isn't increased when calling calculateNextQAVersionDetect. Why would't you want to increase the SIGQA value? (This question applies to some of the tests below)